src/PIL/ContainerIO.py

```diff
@@ -92,13 +92,13 @@ def readline(self):
 
         :returns: An 8-bit string.
         """
-        s = b""
+        s = b"" if "b" in self.fh.mode else ""
         while True:
             c = self.read(1)
             if not c:
                 break
             s = s + c
-            if c == b"\n":
+            if c in [b"\n", "\n"]:
```

Comment:
This line results in a `BytesWarning` when running with the Python `-b` argument:

```
Tests/test_file_tar.py::TestFileTar::test_sanity
.../Pillow/.tox/py38/lib/python3.8/site-packages/PIL/ContainerIO.py:101: BytesWarning: Comparison between bytes and string
    if c in [b"\n", "\n"]:
```

Reply:
Thanks. I've updated the commit.